### PR TITLE
Add support for cloud_displayname

### LINKED ISSUE #965: Support cloud_displayname for saml_token
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritise this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritise the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add support for cloud_displayname as optional claims. See https://learn.microsoft.com/en-us/azure/active-directory/develop/active-directory-optional-claims for the support.

### New or Affected Resource(s)

* azuread_application

### Potential Terraform Configuration

```hcl
  optional_claims {
    saml2_token {
      additional_properties = [
        "cloud_displayname", #<< should be cloud_displayname but currently not supported by the provider
      ]
      essential = true
      name      = "groups"
    }
  }```

### PR DESCRIPTION
**Pull request by cvs79**
Fixes #965

### CODE REVIEW

File: internal/services/applications/migrations/application_resource.go [hunk 1]
```diff
@@ -331,6 +331,7 @@ func ResourceApplicationInstanceResourceV0() *schema.Resource {
 													"netbios_domain_and_sam_account_name",
 													"sam_account_name",
 													"use_guid",
+													"cloud_displayname",
```
Comment (manicminer) [inline internal/services/applications/migrations/application_resource.go:334]: This file contains a representation of the schema from an earlier point in time and should not be updated
```suggestion
```

File: internal/services/applications/migrations/application_resource.go [hunk 2]
```diff
@@ -377,6 +378,7 @@ func ResourceApplicationInstanceResourceV0() *schema.Resource {
...
```
Comment (manicminer) [inline internal/services/applications/migrations/application_resource.go:381]: As above
```suggestion
```

File: internal/services/applications/schema.go [hunk 3]
```diff
@@ -51,6 +51,7 @@ func schemaOptionalClaims() *schema.Schema {
 								"on_premise_security_identifier",
 								"sam_account_name",
 								"use_guid",
+								"cloud_displayname",
```
Comment (manicminer) [inline internal/services/applications/schema.go:54]: Can we keep the (alphabetical) ordering here?

**Review by manicminer — CHANGES_REQUESTED** [review]
Hi @cvs79, thanks for this contribution! I've added some comments inline below, if you can take a look at these then this should be good to merge. Thanks!

File: internal/services/applications/migrations/application_resource.go (hunk 2 above)
Comment (cvs79) [reply internal/services/applications/migrations/application_resource.go]: removed the change

File: internal/services/applications/schema.go (hunk 3 above)
Comment (cvs79) [reply internal/services/applications/schema.go]: Sorted alphabetically

**Comment by cvs79** [comment]
@manicminer fixed the comments but it seems the workflow is failing on something out side of my control now.

**Review by manicminer — APPROVED** [review]
@cvs79 That's great, thanks for updating! The workflow failure is unrelated, we'll fix that separately.

**Pull request merged**